kotlin/src/commonMain/kotlin/com/google/openlocationcode/CharBuffer.kt

```diff
@@ -0,0 +1,24 @@
+package com.google.openlocationcode
```

Comment:
This was originally a `StringBuilder` in Java to expand as needed and allow setting new characters into existing positions.  The common non-JVM lib of Kotlin doesn't have the ability to write at an index into its `StringBuilder` and therefore I created an array wrapper to act like a mini StringBuilder, but I could look at it more carefully and maybe just use a String that is concatenated at each step, but it would be a lot of string copies on each append, and no ability to update existing values.  It maybe just a fixed size `CharArray` is fine, I just didn't go back to dig into the code more.

Reply:
I commented this class and made it internal.  It can be removed in the future when Kotlin common allows conversion of the StringBuilder or String to a `CharArray` without using experimental API.

Reply:
I see that StringBuilder has a [setCharAt](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/-string-builder/set-char-at.html) method. Is there a reason we can't use that?